### PR TITLE
Replaced MySQL 5.7 installed from a PPA with an official Docker image

### PR DESCRIPTION
**Pull request by morozov**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | [job #495144274](https://travis-ci.org/doctrine/dbal/jobs/495144274)

### CODE REVIEW

File: tests/travis/install-mysql-5.7.sh [hunk 1]
```diff
@@ -2,21 +2,14 @@
...
-
-echo "Restart mysql..."
-sudo mysql -e "use mysql; update user set authentication_string=PASSWORD('') where User='root'; update user set plugin='mysql_native_password';FLUSH PRIVILEGES;"
+sudo docker run \
+    -d \
+    -e MYSQL_ALLOW_EMPTY_PASSWORD=yes \
+    -e MYSQL_DATABASE=doctrine_tests \
+    -p 33306:3306 \
+    --name mysql57 \
+    mysql:5.7
```
Comment (SenseException) [inline tests/travis/install-mysql-5.7.sh]: Should a fixed 5.7 version/tag be used here or does the build always have to use the newest 5.7 version? I don't expect problems though.

Comment (morozov) [reply tests/travis/install-mysql-5.7.sh]: We have the same for all other images except for DB2 (not sure if it was tagged deliberately). Unless stability issues start showing up, I think testing against the latest is fine.

**Review by lcobucci — APPROVED** [review]
Looks good, I'd suggest to add a tmpfs mount for the data folder, which makes things slightly faster but it can be done in another PR.

More info: https://vladmihalcea.com/how-to-run-integration-tests-at-warp-speed-with-docker-and-tmpfs/

**Review by Ocramius — APPROVED** [review]
:+1:

**Pull request merged**